truss/templates/control/control/endpoints.py

```diff
@@ -26,7 +25,12 @@ async def proxy(request: Request):
         request.app.state.inference_server_process_controller
     )
     client: httpx.AsyncClient = request.app.state.proxy_client
-    url = URL(path=request.url.path, query=request.url.query.encode("utf-8"))
+
+    path = request.url.path
+    if path == "/v1/models/model":
```

Comment:
need to ensure we still match on `/v1/models/{model_name}`, will update the logic here

Reply:
took a closer look at where we call these inference server endpoints and it looks like we hardcode `v1/models/model` everywhere else, I think its ok hardcoding it here as well.

Reply:
clarifying what the logic here would be:
```
path = request.url.path
path_params = request.path_params["path"].split("/")
if len(path_params) == 2 and path_params[0] == "models":
    # Reroute health checks to the inference server's /v1/models/{model_name}/loaded endpoint
    path += "/loaded"
```
since we technically define `{model_name}` as a path parameter in the inference server, we could support this parameter in the control server routing logic i've added in this PR

Reply:
chatted with @squidarth and went with keeping the checks hardcoded to `v1/models/model` since we use this everywhere we call the inference server, and moved the string comparison logic to a helper fn: `_reroute_if_health_check`